consensus/bor/bor.go

```diff
@@ -464,18 +464,11 @@ func (c *Bor) verifyCascadingFields(chain consensus.ChainHeaderReader, header *t
 
 	// Verify the validator list match the local contract
 	if IsSprintStart(number+1, c.config.CalculateSprint(number)) {
-		parentBlockNumber := number - 1
 
-		var newValidators []*valset.Validator
-
-		var err error
-
-		for newValidators, err = c.spanner.GetCurrentValidators(context.Background(), chain.GetHeaderByNumber(parentBlockNumber).Hash(), number+1); err != nil && parentBlockNumber > 0; {
-			parentBlockNumber--
-		}
+		newValidators, err := c.spanner.GetCurrentValidatorsByBlockNrOrHash(context.Background(), rpc.BlockNumberOrHashWithNumber(rpc.LatestBlockNumber), number+1)
 
 		if err != nil {
-			return errUnknownValidators
+			return err
 		}
 
 		sort.Sort(valset.ValidatorsByAddress(newValidators))
```

Comment:
@0xsharma and me were discussing if "sorting" the validators available locally and those available in header's extra data makes sense or not. Not sure if that was already discussed or not but just posting in here for visibility. 

I think we should compare the validator set as it is without sorting. Wdyt?